src/python/ensembl/io/genomio/manifest/manifest.py

```diff
@@ -0,0 +1,176 @@
+        "genome",
+        "seq_attrib",
+        "seq_region",
+        "agp",
+        "events",
+    }
+    alias_names = {
+        "gene_models": "gff3",
+        "dna": "fasta_dna",
+        "pep": "fasta_pep",
+    }
+    names = {name: name for name in same_names}
+    names = {**names, **alias_names}
+    multi_files = {"agp"}
+
+    def __init__(self, manifest_dir: Path) -> None:
+        """Initializes a manifest with the directory containing the files (and a manifest if it exists).
+
+        Args:
+            manifest_dir: directory where the files are contained.
+        """
+        self.root_dir = manifest_dir
+        self.file_path = manifest_dir / "manifest.json"
+        self.files: dict = {}
+
+    def create(self) -> None:
+        """Creates a manifest file from the files in a directory."""
+        self.get_files_checksums()
+        with self.file_path.open("w") as json_out:
+            json_out.write(json.dumps(self.files, sort_keys=True, indent=4))
+
+    def get_files_checksums(self) -> ManifestDict:
+        """Records all the files in the directory with their checksum."""
+        manifest_files: ManifestDict = {}
+        for subfile in self.root_dir.iterdir():
+            logging.debug(f"Check file {subfile} ({subfile.stem}, {subfile.suffix})")
+            used_file = False
+            if subfile.is_dir():
+                logging.warning("Can't create manifest for subdirectory")
+                continue
+
+            # Delete and skip empty files
+            if subfile.stat().st_size == 0:
+                logging.warning(f"Skip and delete empty file: {subfile}")
+                subfile.unlink()
+                continue
+
+            for name, standard_name in self.names.items():
+                # Either the last element of the stem or the suffix is a known name
+                if subfile.stem.endswith(name) or subfile.suffix == f".{name}":
+                    logging.debug(f"Matched to {name} ({standard_name}) = {subfile}")
+                    used_file = True
+                    md5 = self._get_md5sum(subfile)
+                    file_obj = {"file": subfile.name, "md5sum": md5}
+
+                    # Multiple files stored, each with a name
+                    if standard_name in self.multi_files:
+                        manifest_files.setdefault(standard_name, {})
+                        obj_name = self._prepare_object_name(subfile, name, manifest_files[standard_name])
+                        manifest_files[standard_name][obj_name] = file_obj
+
+                    # Single file/init
+                    else:
+                        manifest_files[standard_name] = file_obj
+
+            if not used_file:
+                logging.warning(f"File {subfile} was not included in the manifest")
+
+        self.files = manifest_files
+        return self.files
+
+    def _prepare_object_name(self, subfile: Path, name: str, manifest_dict: dict[str, dict[str, str]]) -> str:
```

Comment:
Flagging this just in case: It can be a little confusing that the `ManifestDict` type is not applicable for an argument called `manifest_dict`...